include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,844 @@
+ *
+ */
+typedef struct odp_comp_session_param_t {
+	/** Compress vs. Decompress operation */
+	odp_comp_op_t op;
+
+	/** Sync vs Async mode
+	 *
+	 * When mode = ODP_COMP_SYNC, odp_comp_xxx()
+	 * should be called.
+	 *
+	 * When mode = ODP_COMP_ASYNC, odp_comp_xxx_enq()
+	 * should be called.
+	 *
+	 * Use odp_comp_capability() for supported mode.
+	 *
+	 */
+	odp_comp_op_mode_t mode;
+
+	/** Compression algorithm
+	 *
+	 *  Use odp_comp_capability() for supported algorithms.
+	 */
+	odp_comp_alg_t comp_algo;
+
+	/** Hash algorithm
+	 *
+	 *  Use odp_comp_alg_capability() for supported hash algo for
+	 *  compression algo given as comp_alg. Implementation should not
+	 *  support hash only operation on data. output should always contain
+	 *  data + hash.
+	 *
+	 */
+	odp_comp_hash_alg_t hash_algo;
+
+	/** parameters specific to compression */
+	odp_comp_algo_param_t alg_param;
+
+	/** Async mode completion event queue
+	 *
+	 * When mode = ODP_COMP_ASYNC, user should wait on ODP_EVENT_PACKET
+	 * with subtype ODP_EVENT_PACKET_COMP on this queue.
+	 *
+	 * By default, implementation enques completion events in-order-of
+	 * request submission and thus queue is considered ordered.
+	 *
+	 * Please note, behavior could be changed or enhanced
+	 * to queue event in-order-of their completion to enable
+	 * performance-oriented application to leverage hw offered parallelism.
+	 * However, this will be subject to application requirement and more
+	 * explicit defined use-case.
+	 *
+	 */
+	odp_queue_t compl_queue;
+} odp_comp_session_param_t;
+
+/**
+ * Comp API per packet operation result
+ *
+ */
+typedef struct odp_comp_packet_result_t {
+	/** Operation Return Code */
+	odp_comp_err_t err;
+} odp_comp_packet_result_t;
+
+/**
+ * Comp packet API per packet operation parameters
+ */
+typedef struct odp_comp_packet_op_param_t {
+	/** Session handle from creation */
+	odp_comp_session_t session;
+
+	/** Boolean indicating End of data, where
+	 *
+	 *   true : last chunk
+	 *
+	 *   false: more to follow
+	 *
+	 * If set to true, indicates this is the last chunk of
+	 * data which was being processed in stateful mode. After processing
+	 * of last chunk of data is complete (i.e.call return with any
+	 * error code except ODP_COMP_ERR_OUT_OF_SPACE),
+	 * implementation should move algorithm to stateless mode
+	 * for next of batch of operation i.e. reset history,
+	 * insert 'End of Block' marker into compressed data stream, if
+	 * supported by algo.(See deflate/zlib for interpretation of
+	 * stateless/stateful).
+	 *
+	 * For stateless compressions (ex ipcomp), last should be set to 'true'
+	 * for every input packet processing call.
+	 *
+	 * For compression + hash, digest will be available after
+	 * last chunk is processed completely.
+	 */
+	odp_bool_t last;
+
+	/** Input data range */
+	odp_packet_data_range_t indata_range;
+
+	/** Output packet data range. Indicates where
+	* processed packet output will be written.
+	*/
```

Comment:
Probably something is missing here.

Reply:
yes. its an output data range field. will add